pkg/dackbox/indexer/lazy.go

```diff
@@ -83,6 +91,44 @@ func (li *lazyImpl) runIndexing() {
 	}
 }
 
+type hashableField struct {
+	Name           string
+	Value          []byte
+	ArrayPositions []uint64
+}
+
+type hashableDoc struct {
+	ID     string
+	Fields []hashableField
+}
+
+func (li *lazyImpl) evaluateDeduping(doc *document.Document) bool {
+	hashableDoc := hashableDoc{
+		ID:     doc.ID,
+		Fields: make([]hashableField, 0, len(doc.Fields)),
+	}
+	for _, f := range doc.Fields {
```

Comment:
Just curious, is `doc.Fields` order deterministic? I wonder if you really need hashstructure and couldn't just compute the hash on the fly, if performance is a concern. It's just a sequence of writes to the hasher, no?

Reply:
doc.Fields is deterministic
```
		hasher.Write([]byte(doc.ID))
		for _, f := range doc.Fields {
			hasher.Write([]byte(f.Name()))
			hasher.Write(f.Value())
			positionBytes := make([]byte, 8)
			for _, pos := range f.ArrayPositions() {
				binary.LittleEndian.PutUint64(positionBytes, pos)
				hasher.Write(positionBytes)
			}
		}
		hasher.Sum64()
```
Could replace this and be much faster than hashstructure. ~35-40x faster